workspace/dry_clone.py

```
@@ -0,0 +1,35 @@
+    """
+    if not repo_url:
+        return {"ok": False, "error": "no_repo_url"}
+    tmpdir = None
+    try:
+        tmpdir = tempfile.mkdtemp(prefix="preflight-clone-")
+        auth_url = repo_url
+        if token and repo_url.startswith("https://"):
+            auth_url = repo_url.replace("https://", f"https://{token}@")
+        # Use --no-checkout and --depth=1 to minimize network/data
+        cmd = ["git", "clone", "--no-checkout", "--depth", "1", auth_url, tmpdir]
+        proc = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=timeout)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)